### PR TITLE
Feature board

### PR DESCRIPTION
**Pull request by TheREK3R**
Ok, so board kinda works, you can place anything you want but the ai also does and it shows up, ill leave notes in issues panel

### CODE REVIEW

File: src/main/java/cs361/battleships/models/Game.java [hunk 1]
```diff
@@ -24,7 +29,7 @@ public boolean placeShip(Ship ship, int x, char y, boolean isVertical) {
         do {
             // AI places random ships, so it might try and place overlapping ships
             // let it try until it gets it right
-            opponentPlacedSuccessfully = opponentsBoard.placeShip(ship, randRow(), randCol(), randVertical());
+            opponentPlacedSuccessfully = opponentsBoard.placeShip(ship, randCol(), randVertical());
```
Comment (guyera) [inline src/main/java/cs361/battleships/models/Game.java:32]: Why did you delete randRow()? Or am I reading this wrong? The placeShip method requires four parameters, it looks like there are only three. This would not compile, right?

File: src/main/java/cs361/battleships/models/Game.java [hunk 2]
```diff
@@ -4,14 +4,19 @@
...
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Random;
 
 import static cs361.battleships.models.AtackStatus.*;
 
 public class Game {
 
-    @JsonProperty private Board playersBoard = new Board();
-    @JsonProperty private Board opponentsBoard = new Board();
```
Comment (guyera) [inline src/main/java/cs361/battleships/models/Game.java:13]: This was fine as it was. In Java, you can initialize in the member declaration instead of using a default constructor.

File: src/main/java/cs361/battleships/models/Board.java [hunk 3]
```diff
@@ -3,46 +3,72 @@
...
-		// TODO Implement
-		return false;
+        int size = 2;
+        if (ship.getKind().equals("MINESWEEPER")){size = 2;}
+        if (ship.getKind().equals("DESTROYER")){size = 3;}
+        if (ship.getKind().equals("BATTLESHIP")){size = 4;}
+
+        if(isVertical == true){
+            if(x > size){
+                ship.addFeatures(x, y, isVertical, size);
```
Comment (guyera) [inline src/main/java/cs361/battleships/models/Board.java:36]: Should just use a setSize() function, especially since none of the other properties changed

File: src/main/java/cs361/battleships/models/Board.java [hunk 4]
```diff
@@ -3,46 +3,72 @@
...
+                ships.add(ship);
+                return true;
+            }
+        }
+        else if(y <= ('J'- (char)size)){
+            ship.addFeatures(x, y, isVertical, size);
```
Comment (guyera) [inline src/main/java/cs361/battleships/models/Board.java:42]: same here, setSize()

File: src/main/java/cs361/battleships/models/Board.java [hunk 5]
```diff
@@ -3,46 +3,72 @@
...
+        if (ship.getKind().equals("BATTLESHIP")){size = 4;}
+
+        if(isVertical == true){
+            if(x > size){
+                ship.addFeatures(x, y, isVertical, size);
+                ships.add(ship);
+                return true;
+            }
+        }
+        else if(y <= ('J'- (char)size)){
```
Comment (guyera) [inline src/main/java/cs361/battleships/models/Board.java:41]: The math is off here. If the size was 1, y could be equal to j, but this suggests it would have to be less than or equal to 'J' - 1, which is 'I' or anything before.

**Comment by guyera** [comment]
I'll go ahead and merge. There are a few things that need to be changed, but I can change them when implementing the next feature.

**Comment by zhangji52** [comment]
After testing the place ship function in the board.java, it works perfectly fine. Maybe we should add some more packages of modules when we are writing for the attack function, otherwise, it looks really good to me, the function is short without bugs
